cpp/daal/src/algorithms/dtrees/forest/classification/oneapi/df_classification_predict_dense_oneapi_impl.i

```diff
@@ -286,11 +288,32 @@ services::Status PredictKernelOneAPI<algorithmFPType, method>::predictByAllTrees
         if (weighted)
         {
             const double * probas = pModel->getProbas(iTree);
-            context.copy(probasArr, iTree * maxTreeSize * _nClasses, (void *)probas, treeSize * _nClasses, 0, treeSize * _nClasses, status);
-            DAAL_CHECK_STATUS_VAR(status);
+
+            if (sizeof(algorithmFPType) == sizeof(double))
+            {
+                services::internal::tmemcpy<double, sse2>(reinterpret_cast<double *>(probasArrHost.get()) + iTree * maxTreeSize * _nClasses, probas,
+                                                          treeSize * _nClasses);
+            }
+            else
+            {
+                algorithmFPType * dst_ptr = probasArrHost.get() + iTree * maxTreeSize * _nClasses;
+
+                PRAGMA_IVDEP
+                PRAGMA_VECTOR_ALWAYS
+                for (size_t i = 0; i < treeSize * _nClasses; i++)
+                {
+                    dst_ptr[i] = static_cast<algorithmFPType>(probas[i]);
+                }
+            }
         }
     }
 
+    if (weighted)
+    {
+        context.copy(probasArr, 0, (void *)probasArrHost.get(), treeBlockSize * _nClasses, 0, treeBlockSize * _nClasses, status);
+        DAAL_CHECK_STATUS_VAR(status);
+    }
```

Comment:
This branch is not needed, the same condition is above, they can be combined into one

Reply:
This branch is needed to mitigate perf degradation caused by call of context.copy for each tree. Here it is performed only once for tree block, which results in perf improvement up to 20% for hepmass/double case for example.

Reply:
Ok